src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -79,7 +80,9 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs args)
                     // this consumed around 2%-3% of the trace after some other optimizations I did. Most of that
                     // was actually walking the documents list since this was causing all the Documents to be realized.
                     // Since this is on the UI thread, it's best just to not do the work if we don't need it.
-                    if (oldProject.SupportsCompilation && !object.Equals(oldProject.ParseOptions, newProject.ParseOptions))
+                    if (!ServiceFeatureOnOffOptions.IsPowerSaveModeEnabled(newProject) &&
+                        oldProject.SupportsCompilation &&
+                        !object.Equals(oldProject.ParseOptions, newProject.ParseOptions))
```

Comment:
1. need to update this doc.
2. this behavior seems undesirable...  if the project changed and the parse options were updated, it's *really* important to update our syntax trees.  otherwise, all sorts of debu/release stuff will be incorrect. 

This is another case of the user taking an explicit action, so i think it's ok to do work in response to that.  i think we want to avoid the costly stuff the user didn't explicitly ask to have happen.

Reply:
@CyrusNajmabadi Will this feature cause the compilation to be realized? Note that I still need to test the effects of this change thoroughly, but presume you already know.

Reply:
i retract '2'.  We should still get up to date synta xtree, we just won't proactively make it happen.

So i'm ok with this change, but i also think it's *marginal*.  Anything that needs the semantic model (which is like everything) will need up to date trees, so they're going to force those trees to reparse anyways.  

--

Another way of thinking aobut it:  classification is fast and causes semantics (i.e. the compilation) to be computed for the project on each keystroke.  If you work causes us to avoid computation that classification would cause anyways, it's not really doing much afaict.

on the other hand, having us not keep our indices up to date in the background *is* relevant as those may want entire symbol graphs for a project.